zipkin-server/src/main/java/zipkin2/server/internal/prometheus/ZipkinMetricsController.java

```diff
@@ -47,11 +54,13 @@ public ObjectNode fetchMetricsFromMicrometer() {
 
       Meter.Type type = meter.getId().getType();
       if (type == Meter.Type.COUNTER) {
-        metricsJson.put("counter." + name + "." + transport, ((Counter) meter).count());
+        generator.writeNumberField("counter." + name + "." + transport, ((Counter) meter).count());
       } else if (type == Meter.Type.GAUGE) {
-        metricsJson.put("gauge." + name + "." + transport, ((Gauge) meter).value());
+        generator.writeNumberField("gauge." + name + "." + transport, ((Gauge) meter).value());
       } // We only use counters and gauges
     }
-    return metricsJson;
+    generator.writeEndObject();
+    generator.flush();
```

Comment:
Ah hadn't noticed this on the other snippets - I'd usually use try/resources on the generator instead of manually flushing

Reply:
in some cases this helps, though not 100% it is that much more clear. I changed to try/finally where the extra indent level didn't cause more pain